_aggregations/bucket/terms.md

```diff
@@ -58,16 +58,51 @@ GET opensearch_dashboards_sample_data_logs/_search
+
+By default the `shard_size` parameter is set to `size * 1.5 + 10`.
+
+When using concurrent segment search, the `shard_size` parameter is also applied to each segment slice. 
+
+The `shard_size` parameter serves as a way to balance between performance and document count accuracy for terms aggregations. Higher `shard_size` values will ensure higher document count accuracy but with higher memory and compute usage. Lower `shard_size` values will be more performant but with lower document count accuracy.
+
+## Document count error
+
 The response also includes two keys named `doc_count_error_upper_bound` and `sum_other_doc_count`.
 
-The `terms` aggregation returns the top unique terms. So, if the data has many unique terms, then some of them might not appear in the results. The `sum_other_doc_count` field is the sum of the documents that are left out of the response. In this case, the number is 0 because all the unique values appear in the response.
+The `terms` aggregation returns the top unique terms. So, if the data has many unique terms, then some of them might not appear in the results. The `sum_other_doc_count` field is the sum of the documents that are left out of the response. In this case, the number is 0 because all the unique values appear in the response. 
+
+The `doc_count_error_upper_bound` field represents the maximum possible count for a unique value that's left out of the final results. Use this field to estimate the error margin for the count. 
+
+The `doc_count_error_upper_bound` value and the entire notion of accuracy is only applicable to aggregations using the default sort order---by document count, descending. This is because when you sort by descending document count, any terms that were not returned are guaranteed to have equal or fewer documents than those terms which were returned. Based on this, you can compute the `doc_count_error_upper_bound`.
+
+If the `show_term_doc_count_error` parameter is set to `true`, then the `terms` aggregation will show the `doc_count_error_upper_bound` computed for each unique bucket in addition to the overall value.
+
+## The `min_doc_count` and `shard_min_doc_count` parameters
+
+You can use the `min_doc_count` parameter to filter out any unique terms with less than `min_doc_count` results. The `min_doc_count` threshold is applied only after merging the results retrieved from all of the shards. Each shard does not know about the global document count for a given term. Thus, you may get unexpected results when using the `min_doc_count` parameter if there is a big difference between the list of top `shard_size` globally frequent terms and shard-locally frequent terms.
+
+Separately, the `shard_min_doc_count` parameter is used to filter out the unique terms a shard returns back to the coordinator with less than `shard_min_doc_count` results.
+
+When using concurrent segment search, the `shard_min_doc_count` parameter is not applied to each segment slice.
```

Comment:
Do we have to mention this since it will be applied at shard level as part of reduce in concurrent search case ?

Reply:
I think it's good to explicitly call it out here since it's not obvious that `shard_size` would be applied at the slice level but `shard_min_doc_count` is not.

Reply:
I think we can link to https://github.com/opensearch-project/OpenSearch/issues/11847 here

Reply:
@kolchfa-aws would you mind adding the link to the GitHub issue to the docs here for me? I don't want to push a new commit and mess up the suggestions from @natebower below. Thanks!